catroid/src/main/java/org/catrobat/catroid/embroidery/DSTFileGenerator.java

```diff
@@ -0,0 +1,309 @@
+
+	private float minX;
+	private float maxX;
+	private float minY;
+	private float maxY;
+	private File dstFile;
+	private String filename;
+	private FileOutputStream fileStream;
+
+	ArrayList<PointF> stitchPoints = null;
+
+	public DSTFileGenerator(File cache) {
+		filename = ProjectManager.getInstance().getCurrentProject().getName() + ".dst";
+		dstFile = new File(cache, filename);
+
+		if (dstFile.exists()) {
+			dstFile.delete();
+		}
+
+		try {
+			dstFile.createNewFile();
+		} catch (IOException e) {
+			Log.e(TAG, Log.getStackTraceString(e));
+		}
+
+		stitchPoints = new ArrayList<>();
+	}
+
+	public File createDSTFile(ArrayList<PointF> stitchPoints) {
+		if (stitchPoints.size() > 1) {
+			prepareFileWriting(stitchPoints);
+			writeDSTFile();
+			reset();
+			return dstFile;
+		}
+		return null;
+	}
+
+	private void prepareFileWriting(ArrayList<PointF> stitchPoints) {
+		convertStitchPointsUnit(stitchPoints);
+		validateStitches();
+		calculateBoundingBox();
+	}
+
+	private void writeDSTFile() {
+		try {
+			writeHeader();
+			writeStitches();
+			writeEndFile();
+		} catch (IllegalArgumentException ex) {
+			Log.e(TAG, "Remaining stitch distance should be zero!");
+		} catch (IOException e) {
+			Log.e(TAG, Log.getStackTraceString(e));
+		}
+	}
+
+	private void convertStitchPointsUnit(ArrayList<PointF> stitchPoints) {
+		DisplayMetrics metrics = Resources.getSystem().getDisplayMetrics();
+		Project project = ProjectManager.getInstance().getCurrentProject();
+		Float inchMmFactor = 25.4f;
+		Float dotOneMmFactor = 10.0f;
+		Float aspectRatioMultiplierX = (float) metrics.widthPixels / project.getXmlHeader().getVirtualScreenWidth();
+		Float aspectRatioMultiplierY = (float) metrics.heightPixels / project.getXmlHeader().getVirtualScreenHeight();
+
+		if (project.getScreenMode() == ScreenModes.MAXIMIZE) {
+			if (aspectRatioMultiplierX > aspectRatioMultiplierY) {
+				aspectRatioMultiplierX = aspectRatioMultiplierY;
+			} else {
+				aspectRatioMultiplierY = aspectRatioMultiplierX;
+			}
+		}
+
+		for (PointF stitch: stitchPoints) {
+			PointF stitchUnitMm = new PointF();
+			stitchUnitMm.x = stitch.x * inchMmFactor * dotOneMmFactor * aspectRatioMultiplierX / metrics.densityDpi;
+			stitchUnitMm.y = stitch.y * inchMmFactor * dotOneMmFactor * aspectRatioMultiplierY / metrics.densityDpi;
+			this.stitchPoints.add(stitchUnitMm);
+		}
+	}
+
+	private void validateStitches() {
+		int xDistance;
+		int yDistance;
+
+		for (int index = 0; index < stitchPoints.size() - 1; index++) {
+			PointF currentPoint = stitchPoints.get(index);
+			PointF nextPoint = stitchPoints.get(index + 1);
+
+			xDistance = (int) (nextPoint.x - currentPoint.x);
+			yDistance = (int) (nextPoint.y - currentPoint.y);
+
+			if ((xDistance < -MAXDISTANCE) || (xDistance > MAXDISTANCE)
+					|| (yDistance < -MAXDISTANCE) || (yDistance > MAXDISTANCE)) {
+				splitStitchAtIndex(index);
+			}
+		}
+	}
+
+	private void splitStitchAtIndex(int index) {
```

Comment:
since this doesnt split the stitch path but interpolates additional stitches inbetween, please rename accordingly.